hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/TssBaseServiceImpl.java

```diff
@@ -388,4 +396,41 @@ public void manageTssStatus(final State state) {
     public TssKeysAccessor getTssKeysAccessor() {
         return tssKeysAccessor;
     }
+
+    public void processTssEncryptionKeyChecks(
+            @NonNull final UserTxn userTxn,
+            @NonNull final HandleContext handleContext,
+            @NonNull final KeysAndCerts keysAndCerts) {
+        final var readableStoreFactory = new ReadableStoreFactory(userTxn.state());
+        final var tssStore = readableStoreFactory.getStore(ReadableTssStore.class);
+        final var tssEncryptionKeys = tssStore.getTssEncryptionKeys(
+                handleContext.networkInfo().selfNodeInfo().nodeId());
+        Duration timeSinceLastSubmission = tssSubmissions.getLastSuccessfulTssEncryptionKeySubmission() == null
+                ? null
+                : Duration.between(
+                        tssSubmissions.getLastSuccessfulTssEncryptionKeySubmission(), userTxn.consensusNow());
+        final var tssEncryptionKeyRetryDelay =
+                handleContext.configuration().getConfigData(TssConfig.class).tssEncryptionKeyRetryDelay();
+        final var tssEncryptionKeySubmissionRetries =
+                handleContext.configuration().getConfigData(TssConfig.class).tssEncryptionKeySubmissionRetries();
+        if ((tssEncryptionKeys == null
```

Comment:
Perhaps we should also only submit our TSS public key if we are a part of the current active/candidate roster? What do others think?

Reply:
In the `manageTssStatus` state machine, we look at the roster we are trying to key and see if the entries in that roster have encryption keys. If we move this functionality inside `manageTssStatus` then it already looks at the roster being keyed.

Reply:
Moved processTssEncryptionKeyChecks to manageTssStatus